src/main/webapp/src/components/ratingList.js

```diff
@@ -31,71 +33,74 @@ function EnumeratedList(props) {
         {index + 1}
       </ListItemIcon>
       <ListItemText
-        primary={item.songName}
-        secondary={item.bandName + '. From album "' + item.albumName + '"'}
+        primary={item.name}
+        secondary={item.artist}
       />
     </ListItem>
   ));
   return <List>{reactListItems}</List>;
 }
 
 class RatingList extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      topSongs: [],
+      isLoading: false,
+      error: null,
+    };
+  }
+
+  componentDidMount() {
+    this.setState({ isLoading: true });
+
+    axios
+      .get("/top")
+      .then((result) => result.data)
+      .then((topSongs) =>
+        this.setState({
+          topSongs: topSongs,
+          isLoading: false,
+        })
+      )
+      .catch((error) =>
+        this.setState({
+          error,
+          isLoading: false,
+        })
+      );
+  }
+
   render() {
     const classes = this.props.classes;
+    const topSongs = this.state.topSongs;
+
+    if (this.state.error) {
+      return (
+        <div>
+          <p>{this.state.error.message}</p>
```

Comment:
Can you give an example of error? I'm thinking that maybe the user will have no idea what it actually says

Reply:
It depends on the servlet actually. Usually, it displays error that the servlet can send.

Reply:
Maybe we could print this error to console so we know what's happening, but show to the user only something like
`Oops, something went wrong, please try again`